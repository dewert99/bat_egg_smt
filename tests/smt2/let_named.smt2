(declare-const a Bool)
(declare-const b Bool)
(declare-fun f (Bool) Bool)
(assert (and (! a :named x) (let ((x b)) x) (f x)))
(check-sat)
(get-model)
(assert (not (and a b (f a))))
(check-sat)