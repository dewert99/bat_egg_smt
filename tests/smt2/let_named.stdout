sat
(
 (define-fun a () Bool true)
 (define-fun b () Bool true)
 (define-fun f ((x!0 Bool)) Bool
  (ite (= x!0 true) true
   false))
 (define-fun x () Bool true)
)
unsat
